pkg/operator/controller/crl/crl_configmap.go

```diff
@@ -0,0 +1,295 @@
+		if err != nil {
+			return false, nil, fmt.Errorf("client CA configmap has an invalid certificate: %w", err)
+		}
+		subjectKeyId := hex.EncodeToString(cert.SubjectKeyId)
+		if len(cert.CRLDistributionPoints) == 0 {
+			continue
+		}
+		if crl, ok := crls[subjectKeyId]; ok {
+			if crl.HasExpired(now) {
+				log.Info("certificate revocation list has expired", "subject key identifier", subjectKeyId)
+			} else {
+				subjectKeyIds = append(subjectKeyIds, subjectKeyId)
+				continue
+			}
+		}
+		log.Info("retrieving certificate revocation list", "subject key identifier", subjectKeyId)
+		if crl, err := getCRL(cert.CRLDistributionPoints); err != nil {
+			// Creating or updating the configmap with incomplete
+			// data would compromise security by potentially
+			// permitting revoked certificates.
+			return false, nil, fmt.Errorf("failed to get certificate revocation list for certificate key %s: %w", subjectKeyId, err)
+		} else {
+			crls[subjectKeyId] = crl
+			subjectKeyIds = append(subjectKeyIds, subjectKeyId)
+		}
+	}
+
+	if len(subjectKeyIds) == 0 {
+		return false, nil, nil
+	}
+
+	buf := &bytes.Buffer{}
+	for _, subjectKeyId := range subjectKeyIds {
+		asn1Data, err := asn1.Marshal(*crls[subjectKeyId])
+		if err != nil {
+			return false, nil, fmt.Errorf("failed to encode ASN.1 for CRL for certificate key %s: %w", subjectKeyId, err)
+		}
+		block := &pem.Block{
+			Type:  "X509 CRL",
+			Bytes: asn1Data,
+		}
+		if err := pem.Encode(buf, block); err != nil {
+			return false, nil, fmt.Errorf("failed to encode PEM for CRL for certificate key %s: %w", subjectKeyId, err)
+		}
+	}
+	crlData := buf.String()
+
+	crlConfigmapName := controller.CRLConfigMapName(ic)
+	crlConfigmap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      crlConfigmapName.Name,
+			Namespace: crlConfigmapName.Namespace,
+		},
+		Data: map[string]string{
+			"crl.pem": crlData,
+		},
+	}
+	crlConfigmap.SetOwnerReferences([]metav1.OwnerReference{ownerRef})
+
+	return true, &crlConfigmap, nil
+}
+
+// getCRL gets a certificate revocation list using the provided distribution
+// points and returns the certificate list.
+func getCRL(distributionPoints []string) (*pkix.CertificateList, error) {
+	var errs []error
+	for _, distributionPoint := range distributionPoints {
+		// The distribution point is typically a URL with the "http"
+		// scheme.  "https" is generally not used because the
+		// certificate list is signed, and because using TLS to get the
+		// certificate list could introduce a circular dependency
+		// (cannot use TLS without the revocation list, and cannot get
+		// the revocation list without using TLS).
+		//
+		// TODO Support ldap.
+		if !strings.HasPrefix(distributionPoint, "http:") {
+			errs = append(errs, fmt.Errorf("unsupported distribution point type: %s", distributionPoint))
+			continue
+		}
+		log.Info("retrieving CRL distribution point", "distribution point", distributionPoint)
+		resp, err := http.Get(distributionPoint)
```

Comment:
Do we need to take into account cluster-wide proxy in this case? (I ask because im really not sure).

Reply:
Should we consider tweaking any of the default settings on the default client (e.g. timeouts)?

Reply:
Good questions.  I guess there's no reason this shouldn't use the proxy, right?  If it's an internal server, the user could just add it to `noProxy`.

Reply:
Makes sense to me.

Reply:
https://github.com/openshift/cluster-ingress-operator/pull/334/commits/ead9d30aca702ff0a0baf254bd3ae4fe22a22048 already added the `config.openshift.io/inject-proxy: ingress-operator` annotation to the operator deployment, and this is an HTTP (not TLS) client, so it should already pick up any cluster proxy configuration.